--- /usr/sbin/syncoid	2020-11-24 13:47:00.000000000 +0300
+++ /usr/sbin/syncoid2	2022-05-30 21:00:01.663485465 +0300
@@ -39,15 +39,15 @@
 		exit 127;
 	}
 
-	if (defined $args{'recursive'}) {
-		foreach my $option(@sendoptions) {
-			if ($option->{option} eq 'R') {
-				warn "invalid argument combination, zfs send -R and --recursive aren't compatible!";
-				pod2usage(2);
-				exit 127;
-			}
-		}
-	}
+#	if (defined $args{'recursive'}) {
+#		foreach my $option(@sendoptions) {
+#			if ($option->{option} eq 'R') {
+#				warn "invalid argument combination, zfs send -R and --recursive aren't compatible!";
+#				pod2usage(2);
+#				exit 127;
+#			}
+#		}
+#	}
 }
 
 my @recvoptions = ();
